File: GRID.py
for _ in range(int(input())):
	n = int(input())
	ls = [input() for i in range(n)]
	x = [True] * n
	y = [True] * n
	ans = 0
	for i in range(n-1, -1, -1):
		for j in range(n-1, -1, -1):
			if ls[i][j] == "#":
				x[i] = False
				y[j] = False
			elif x[i] and y[j]:
				ans += 1
	print(ans)

File: GRID.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main(void) {
	ios::sync_with_stdio(false);
	cin.tie(0);
	cout.tie(0);

#ifndef ONLINE_JUDGE
	freopen("input.txt", "r", stdin);
//	freopen("output.txt", "w", stdout);
#endif

	int t;
	cin >> t;
	while (t--) {
		int n, ans = 0;
		cin >> n;
		vector<string> str(n);
		for (int i = 0; i < n; i++)
			cin >> str[i];
		vector<bool> x(n, true), y(n, true);
		for (int i = n-1; i >= 0; i--)
			for (int j = n-1; j >= 0; j--) {
				if (str[i][j] == '#')
					x[i] = y[j] = false;
				else if (x[i] && y[j])
					ans++;
			}
		cout << ans << endl;
	}
}
